File: services/ingestion-openmeteo/app/Application/Weather/Contracts/LocationRepository.php
<?php

namespace App\Application\Weather\Contracts;

use App\Domain\Weather\Location;

/**
 * Repository interfejs za rad sa lokacijama koje pratimo.
 */
interface LocationRepository
{
    /**
     * Vraća sve aktivne lokacije koje treba uključiti u ingestion proces.
     *
     * @return Location[]
     */
    public function getActiveLocations(): array;
}

File: services/ingestion-weatherapi/internal/application/weather/contracts/location_repository.go
package contracts

import (
	"context"

	"ingestion-weatherapi/internal/domain/weather"
)

/*
LocationRepository predstavlja application contract za pristup aktivnim lokacijama.

Application sloj ne treba da zna da li lokacije dolaze iz Postgresa,
nekog fajla, cache-a ili nečeg trećeg.
*/
type LocationRepository interface {
	/*
		GetActiveLocations vraća sve aktivne lokacije koje treba uključiti u ingestion proces.
	*/
	GetActiveLocations(ctx context.Context) ([]weather.Location, error)
}
